sdk/src/cache/messages/data/scalar/get_batch.rs

```diff
@@ -0,0 +1,221 @@
+                items: get_requests,
+            },
+        )?;
+
+        let mut response_stream = cache_client
+            .next_data_client()
+            .get_batch(get_batch_request)
+            .await?
+            .into_inner();
+
+        // receive stream of get responses
+        let mut results_list: Vec<GetResponse> = vec![];
+        while let Some(get_response) = response_stream.message().await? {
+            let sdk_get_response = match get_response.result() {
+                ECacheResult::Hit => GetResponse::Hit {
+                    value: Value {
+                        raw_item: get_response.cache_body,
+                    },
+                },
+                ECacheResult::Miss => GetResponse::Miss,
+                _ => {
+                    return Err(MomentoError::unknown_error(
+                        "GetBatch",
+                        Some(format!("{:#?}", get_response)),
+                    ))
+                }
+            };
+            results_list.push(sdk_get_response);
+        }
+
+        let results_map: HashMap<String, GetResponse> = string_keys
+            .clone()
+            .into_iter()
+            .zip(results_list.clone())
+            .collect();
+
+        Ok(GetBatchResponse {
+            results: results_list,
+            results_dictionary: results_map,
+        })
+    }
+}
+
+/// Response for a cache get batch operation.
+///
+/// You can use `into()` to convert a `GetBatchResponse` into a `Vec<GetResponse>`, a `HashMap<String, GetResponse>`, a `HashMap<String, Value>`, or a `HashMap<String, Vec<u8>>`.
+/// You can use `try_into()` to convert a `GetBatchResponse` into a `HashMap<String, String>`.
+/// ```
+/// # fn main() -> anyhow::Result<()> {
+/// # use momento_test_util::create_doctest_cache_client;
+/// # tokio_test::block_on(async {
+/// # let (cache_client, cache_name) = create_doctest_cache_client();
+/// use std::convert::TryInto;
+/// use std::collections::HashMap;
+/// use momento::cache::GetResponse;
+/// use momento::cache::messages::data::scalar::get::Value;
+///
+/// let keys = vec!["key1", "key2", "key3"];
+///
+/// let result_list: Vec<GetResponse> = cache_client.get_batch(&cache_name, keys.clone()).await?.into();
+///
+/// let result_map: HashMap<String, GetResponse> = cache_client.get_batch(&cache_name, keys.clone()).await?.into();
+///
+/// let result_map_values: HashMap<String, Value> = cache_client.get_batch(&cache_name, keys.clone()).await?.into();
+///
+/// let result_map_string_values: HashMap<String, Vec<u8>> = cache_client.get_batch(&cache_name, keys.clone()).await?.into();
+///
+/// let result_map_byte_values: HashMap<String, String> = cache_client.get_batch(&cache_name, keys.clone()).await?.try_into().expect("expecting strings");
+/// # Ok(())
+/// # })
+/// # }
+/// ```
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct GetBatchResponse {
+    results: Vec<GetResponse>,
+    results_dictionary: HashMap<String, GetResponse>,
```

Comment:
My preference would be to store only one of these and build the other on the fly. Ie either only a `Vec` or a `HashMap`.

Reply:
I opted to remove the `Vec<GetResponse>` and `Vec<SetResponse>` options instead. `HashMap.iter()` will traverse the items in an arbitrary order, not the order in which the batch of items was received. I think having a direct mapping from key <-> response is clearer.